packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,166 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+
+import 'basic_types.dart';
+import 'text_style.dart';
+import 'text_painter.dart';
+
+/// Called on each span as [InlineSpan.visitChildren] walks the [InlineSpan] tree.
+///
+/// Returns true when the walk should continue, and false to stop visiting further
+/// [InlineSpan]s.
+typedef InlineSpanVisitor = bool Function(InlineSpan span);
+
+/// An immutable span of inline content which forms a paragraph.
+///
+///  * The subclass [TextSpan] specifies text and may contain child [InlineSpan]s.
+///  * The subclass [PlaceholderSpan] represents a placeholder that may be
+///    filled with non-text content. [PlaceholderSpan] itself defines a
+///    [ui.PlaceholderAlignemnt] and a [TextBaseline]. To be useful,
+///    [PlaceholderSpan] should be extended to define content. An instance of
+///    this is the [WidgetSpan] class in the widgets library.
+///  * The subclass [WidgetSpan] specifies embedded inline widgets. Specify an
+///    inline widget by wrapping the widget with a [WidgetSpan].
+///
+/// {@tool sample}
+///
+/// This example shows a tree of [InlineSpan]s that make a query asking for a
+/// name with a [TextField] embedded inline.
+///
+/// ```dart
+/// RichText(
```

Comment:
we should use `Text.rich` or whatever it's called, rather than `RichText`.

Reply:
Just for future reference, is Text.rich the more canonical way to use a RichText? Are there situations where we prefer a raw RichText over Text.rich?

Reply:
`RichText` should really be named "RawText", it's the low-level render object widget that interfaces with RenderParagraph.
`Text` is the high-level widget that honors `DefaultTextStyle`, implements accessibility font scaling, etc.
Generally you only use RichText when you want direct precise control over the pixels.